### PR TITLE
add option in ae/vae to compare output of the decoder with a different target

### PR DESCRIPTION
**Pull request by luigibonati**
## Description
I added a small feature to the training step of AE/VAE cvs to allow comparing the output of the decoder not only with the input data but also with different quantities (e.g. time-lagged data).

If a 'target' key is available in the dictionary it will be used in the reconstruction loss, otherwise 'data' is used.

@andrrizzi

## Status
- [X] Ready to go

### CODE REVIEW

File: mlcvs/cvs/unsupervised/vae.py [hunk 1]
```diff
@@ -182,13 +182,17 @@ def training_step(self, train_batch, batch_idx):
...
 
+        # Reference output (compare with a 'target' key if any, otherwise with input 'data')
+        if 'target' in train_batch:
+            x_ref = train_batch['target']
+        else:
+            x_ref = x 
+
         # Loss function.
-        loss = self.loss_fn(x, x_hat, mean, log_variance, **options)
+        loss = self.loss_fn(x_hat, x_ref, mean, log_variance, **options)
```
Comment (andrrizzi) [inline mlcvs/cvs/unsupervised/vae.py:195]: I'd keep the order ``(x_ref, x_hat, ...)`` here.

The order doesn't matter when the decoder's distribution is a Gaussian like the default (which gives rise to the MSE loss), but when the decoder's distribution is not isotropic, the datapoint (``x_ref``) and the distribution's parameters (``x_hat``) are not usually interchangeable.

**Review by andrrizzi — APPROVED** [review]
Nice!
